feat: improve transaction support and add new endpoints

impredmet opened the pull request:
- Added `.yarn` and `yarn.lock` to `.gitignore`
- Introduced `dev` script for easier development
- Removed `/utxos` endpoint (replaced by `/address/:address/utxos` and `/address/:address/utxos/:amount`)
- Implemented `getTransactionsFromBlock()` in the indexer for comprehensive transaction indexing
- Deleted unused file `storage.ts`
- Changed UTXO amount type to `string` instead of `BigInt`
- Added `TransactionAttributes` to `types.ts`
- Created new `Transaction` model
- Added new `/address` endpoint with subroutes for balance, transactions, and UTXOs

impredmet committed:
feat: improve transaction support and add new endpoints

- Added `.yarn` and `yarn.lock` to `.gitignore`
- Introduced `dev` script for easier development
- Removed `/utxos` endpoint (replaced by `/address/:address/utxos` and `/address/:address/utxos/:amount`)
- Implemented `getTransactionsFromBlock()` in the indexer for comprehensive transaction indexing
- Deleted unused file `storage.ts`
- Changed UTXO amount type to `string` instead of `BigInt`
- Added `TransactionAttributes` to `types.ts`
- Created new `Transaction` model
- Added new `/address` endpoint with subroutes for balance, transactions, and UTXOs

encloinc commented:
For the Transaction model, you need to create BTREE indicies for addresses - otherwise every lookup on db will be O(n). Check out how I did it on the UTXO model.

also, you should add an incoming/spending boolean flag on a Transaction - so on the wallet you can display which transactions are receiving/spending (just like the og api did).

impredmet committed:
feat: add BTREE indexes for TransactionModel

- Added BTREE indexes on `vin` and `vout` fields to optimize address lookups in the Transaction model.

impredmet commented:
> For the Transaction model, you need to create BTREE indicies for addresses - otherwise every lookup on db will be O(n). Check out how I did it on the UTXO model.
> 
> also, you should add an incoming/spending boolean flag on a Transaction - so on the wallet you can display which transactions are receiving/spending (just like the og api did).

The `incoming/spending` flag has been implemented on the wallet side. The previous API did not include this feature.
